docs/ref/models/database-functions.txt

```diff
@@ -293,3 +293,391 @@ Usage example::
+    >>> from django.db.models.functions import (
+        ExtractYear, ExtractMonth, ExtractDay, ExtractWeekDay,
+        ExtractHour, ExtractMinute, ExtractSecond,
+    )
+    >>> start_2015 = datetime(2015, 6, 15, 23, 30, 1, tzinfo=timezone.utc)
+    >>> end_2015 = datetime(2015, 6, 16, 13, 11, 27, tzinfo=timezone.utc)
+    >>> Experiment.objects.create(
+    ...    start_time=start_2015, start_date=start_2015.date(),
+    ...    end_time=end_2015, end_date=end_2015.date())
+    >>> Experiment.objects.annotate(
+    ...     year=ExtractYear('start_time'),
+    ...     month=ExtractMonth('start_time'),
+    ...     day=ExtractDay('start_time'),
+    ...     weekday=ExtractWeekDay('start_time'),
+    ...     hour=ExtractHour('start_time'),
+    ...     minute=ExtractMinute('start_time'),
+    ...     second=ExtractSecond('start_time')
+    ... ).values('year', 'month', 'day', 'weekday', 'hour', 'minute', 'second'
+    ... ).get(end_time__year=ExtractYear('start_time'))
+    {
+        'year': 2015, 'month': 6, 'day': 15, 'weekday': 2,
+        'hour': 23, 'minute': 30, 'second': 1
+    }
+
+When :setting:`USE_TZ` is ``True`` then datetimes are stored in the database
+in UTC. If a user has a different timezone activated, the result will first
+be converted into that timezone before the value is extracted. In the example
+below we convert to the Melbourne timezone which is UTC +10:00, which changes
+the day, weekday, and hour values returned::
+
+    >>> import pytz
+    >>> tzinfo = pytz.timezone('Australia/Melbourne')  # UTC+10:00
+    >>> with timezone.override(tzinfo):
+    ...    Experiment.objects.annotate(
+    ...        day=ExtractDay('start_time'),
+    ...        weekday=ExtractWeekDay('start_time'),
+    ...        hour=ExtractHour('start_time')
+    ...    ).values('day', 'weekday', 'hour'
+    ...    ).get(end_time__year=ExtractYear('start_time'))
+    {
+        'day': 16, 'weekday': 3, 'hour': 9
+    }
+
+``Trunc``
+---------
+
+.. class:: Trunc(expression, kind, output_field=None, tzinfo=None, **extra)
+
+Truncates a date up to a significant component.
+
+When you only care if something happened in a particular year, hour, or day,
+but not the exact second, then ``Trunc`` (and its subclasses) can be useful to
+filter or aggregate your data. Calculating the number of sales per day is a
+common use case for ``Trunc``.
+
+``Trunc`` takes a single ``expression``, representing a ``DateField`` or
+``DateTimeField``, a ``kind`` representing a date part, and an ``output_field``
+that's either ``DateTimeField()`` or ``DateField()``. It returns a datetime or
+date, depending on ``output_field``, with fields up to ``kind`` set to their
+minimum value. If ``output_field`` is omitted, it will default to the
+``output_field`` of ``expression``. A ``tzinfo`` subclass, usually provided by
+``pytz``, can be passed to truncate a value in a specific timezone.
+
+Given the datetime `2015-06-15 14:30:50.000321+00:00` the ``kind``\s supported
```

Comment:
This datetime has more significant digits in the fractional seconds part that the other "Given the datetime" above. If this isn't important or intentional, please make them consistent.